src/compiler/sys.ts

```diff
@@ -1626,14 +1623,10 @@ export let sys: System = (() => {
         };
         return nodeSystem;
 
-        /**
-         * `throwIfNoEntry` was added so recently that it's not in the node types.
-         * This helper encapsulates the mitigating usage of `any`.
-         * See https://github.com/nodejs/node/pull/33716
-         */
+        /** Calls fs.statSync, returning undefined if any errors are thrown */
         function statSync(path: string): import("fs").Stats | undefined {
-            // throwIfNoEntry will be ignored by older versions of node
-            return (_fs as any).statSync(path, { throwIfNoEntry: false });
+            // throwIfNoEntry is available in Node 14.17 and above, which matches our supported range.
+            return _fs.statSync(path, { throwIfNoEntry: false });

```

Comment:
Potentially I can leave a try/catch here just in case someone uses something before 14.17; it wouldn't be faster but probably best overall since all supported users will stop needing to modify the stack lengh limit.